lib/util/Components.js

```diff
@@ -347,12 +347,13 @@ function componentRule(rule, context) {
         var isFunction = /Function/.test(node.type); // Functions
         var isMethod = node.parent && node.parent.type === 'MethodDefinition'; // Classes methods
         var isArgument = node.parent && node.parent.type === 'CallExpression'; // Arguments (callback, etc.)
+        var isJSX = node.parent && node.parent.type === 'JSXExpressionContainer';
```

Comment:
I might be misunderstanding what this is checking for here, but I'm not sure this variable name makes sense. I think `isJSXExpressionContainerExpression` would be more accurate.

Additionally, it might be nice to include a comment with an example of what type of code this identifies.